ls1intum/artemis-android
# `Chore`: Prepare websocket for compressed payload

### Problem Description
To reduce the load on the server with sent websocket updates, the Artemis team decided to compress the payloads of websocket updates if the exceed a certain threshold.

This PR closes #195 


### Changes
 
- Added check for the websocket compression header and optionally decompress the websocket payload
- Added tests for both compressed and normal payloads


### Steps for testing
As the compression is not yet live on any servers, we can only test that un-compressed payloads work as before. 
- Go to any chat with the app
- Write a message to that chat from the webapp
- Observe that the new message is immediately displayed on the app (without the need for a refresh)
